### PR TITLE
Report UnknownTemplateIds warning before sending an error

### PR DESCRIPTION
**Pull request by leo-da**
if no template IDs resolved, send warning before the error, this is for
consistency with the case when at least one ID resolved (we continue
processing request in this case).

Related to: #4521

This PR addresses step 3 from above ticket/discussion:

> Would it make sense to send warnings for all template ids that cannot be resolved and send an error if the list of template ids that can be resolved is empty? Thus, a query containing only invalid template ids would first have a warning for each template id that it cannot be resolved and then an error that there are no template ids that can be resolved. Sending an empty list of template ids would not trigger the warning but only the error.

### Pull Request Checklist

- [ ] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/master/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: ledger-service/http-json/src/test/scala/com/digitalasset/http/WebsocketServiceIntegrationTest.scala [hunk 1]
```diff
@@ -106,6 +106,44 @@ class WebsocketServiceIntegrationTest
...
+  List(
+    SimpleScenario(
+      "query",
+      Uri.Path("/v1/stream/query"),
+      Source.single(TextMessage.Strict("""{"templateIds": ["AA:BB"]}"""))),
+    SimpleScenario(
+      "fetch",
+      Uri.Path("/v1/stream/fetch"),
+      Source.single(TextMessage.Strict("""[{"templateId": "AA:BB", "key": ["k", "v"]}]""")))
+  ).foreach { scenario =>
```
Comment (S11001001) [inline ledger-service/http-json/src/test/scala/com/digitalasset/http/WebsocketServiceIntegrationTest.scala]: Use `TableDrivenPropertyChecks` and `forEvery` for this kind of testing setup. That will still discriminate successful and failing cases without having to interpolate test names.

File: ledger-service/http-json/src/main/scala/com/digitalasset/http/WebSocketService.scala [hunk 2]
```diff
@@ -358,14 +358,12 @@ class WebSocketService(
...
-                implicit val SQ: StreamQuery[q] = qq.alg
-                getTransactionSourceForParty[q](jwt, jwtPayload.party, offPrefix, qq.q: q)
-            }
-            .fold(e => Source.single(-\/(e)), s => s.map(\/-(_))): Source[Error \/ Message, NotUsed]
+        2, // 2 streams max, the 2nd is to be able to send an error back
+        _.map {
+          case (offPrefix, qq: Q.Query[q]) =>
+            implicit val SQ: StreamQuery[q] = qq.alg
+            getTransactionSourceForParty[q](jwt, jwtPayload.party, offPrefix, qq.q: q)
+        }.fold(e => Source.single(-\/(e)), identity): Source[Error \/ Message, NotUsed]
```
Comment (S11001001) [inline ledger-service/http-json/src/main/scala/com/digitalasset/http/WebSocketService.scala]:
```suggestion
        }.valueOr(e => Source.single(-\/(e))): Source[Error \/ Message, NotUsed]
```

File: ledger-service/http-json/src/test/scala/com/digitalasset/http/json/JsonProtocolTest.scala [hunk 3]
```diff
@@ -94,6 +100,34 @@ class JsonProtocolTest
...
+      val expected = JsObject("unknownParties" -> JsArray(expectedParties))
+      x.toJson.asJsObject shouldBe expected
+    }
+    "roundtrips" in forAll(genServiceWarning) { x =>
+      x.toJson.convertTo[domain.ServiceWarning] === x
+    }
+  }
+
+  "domain.WarningsWrapper" - {
+    "WarningsWrapper serialization" in forAll(genWarningsWrapper) { x =>
```
Comment (S11001001) [inline ledger-service/http-json/src/test/scala/com/digitalasset/http/json/JsonProtocolTest.scala]:
```suggestion
    "serialization" in forAll(genWarningsWrapper) { x =>
```

File: ledger-service/http-json/src/test/scala/com/digitalasset/http/WebsocketServiceIntegrationTest.scala (hunk 1 above)
Comment (leo-da) [reply ledger-service/http-json/src/test/scala/com/digitalasset/http/WebsocketServiceIntegrationTest.scala]: `TableDrivenPropertyChecks` and `GeneratorDrivenPropertyChecks` do not work with `AsyncFreeSpec`
```
 found   : Unit
 required: scala.concurrent.Future[org.scalatest.compatible.Assertion]
```

Comment (S11001001) [reply ledger-service/http-json/src/test/scala/com/digitalasset/http/WebsocketServiceIntegrationTest.scala]: Oh well.

Comment (leo-da) [reply ledger-service/http-json/src/test/scala/com/digitalasset/http/WebsocketServiceIntegrationTest.scala]: yeah, it is either `AsyncFreeSpec` or `org.scalatest.concurrent.Futures#whenReady` at the end of every test scenario.

**Pull request merged**